app/view_models/verify/base.rb

```diff
@@ -0,0 +1,88 @@
+module Verify
+  class Base
+    include Rails.application.routes.url_helpers
+
+    def initialize(error: nil, remaining_attempts:)
```

Comment:
Adding named params was very helpful in this refactor! I kept putting args in the wrong order when they were unnamed, resulting in weird and confusing errors. Harder to mess up when there are names!